### ISSUE TITLE
ArgumentError: same file

### ISSUE THREAD

**Issue by elmerland** [imathis/guard-jekyll-plus#38]
I just update from version 2.0.0 to 2.0.1 and it started giving me the error above. Below is the full error dump:

```
17:22:21 - INFO - Reloading browser: gnu_manual.html
17:22:54 - INFO - Jekyll updating files
|
17:22:54 - ERROR - Guard::JekyllPlus failed to achieve its <run_on_modifications>, exception was:
> [#] ArgumentError: same file: js/gnu_manual_script.js and js/gnu_manual_script.js
> [#] /Users/elmer/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:1569:in `block in fu_each_src_dest'
> [#] /Users/elmer/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:1586:in `fu_each_src_dest0'
> [#] /Users/elmer/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:1568:in `fu_each_src_dest'
> [#] /Users/elmer/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:401:in `cp'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-jekyll-plus-2.0.1/lib/guard/jekyll_plus/builder/action.rb:97:in `copy'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-jekyll-plus-2.0.1/lib/guard/jekyll_plus/builder/modifier.rb:18:in `block in do_update'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-jekyll-plus-2.0.1/lib/guard/jekyll_plus/builder/modifier.rb:18:in `each'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-jekyll-plus-2.0.1/lib/guard/jekyll_plus/builder/modifier.rb:18:in `do_update'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-jekyll-plus-2.0.1/lib/guard/jekyll_plus/builder/action.rb:82:in `update'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-jekyll-plus-2.0.1/lib/guard/jekyll_plus/builder.rb:38:in `modified'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-jekyll-plus-2.0.1/lib/guard/jekyll_plus.rb:43:in `run_on_modifications'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:82:in `block in _supervise'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:79:in `catch'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:79:in `_supervise'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:61:in `block (3 levels) in run_on_changes'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:56:in `each'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:56:in `block (2 levels) in run_on_changes'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:119:in `block (2 levels) in _run_group_plugins'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:117:in `each'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:117:in `block in _run_group_plugins'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:116:in `catch'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:116:in `_run_group_plugins'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:54:in `block in run_on_changes'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:53:in `each'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/runner.rb:53:in `run_on_changes'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/internals/queue.rb:23:in `process'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/commander.rb:43:in `start'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/cli/environments/valid.rb:16:in `start_guard'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/cli.rb:113:in `start'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/aruba_adapter.rb:32:in `execute'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/lib/guard/aruba_adapter.rb:19:in `execute!'
> [#] /Users/elmer/.rvm/gems/ruby-2.2.1/gems/guard-2.12.5/bin/_guard-core:11:in `<main>'
17:22:55 - INFO - Guard::JekyllPlus has just been fired
17:22:55 - INFO - Reloading browser: js/gnu_manual_script.js
```

I have teste the following:
- When I change a file in the top level directory everything works fine (e.g. ./my_page.html)
- When changing a file that is one level deep I get the error (e.g. ./js/my_script.js)
- In addition to the error, the process is left running in the background. This means that if I try to run it again it will complain about the port number already been in use.
- When going back to version 2.0.0 I get no such errors. I did not change anything between the two versions, except for changing `guard 'jekyll-plus', {:serve => true} do` to `guard 'jekyll_plus', {:serve => true} do`

**Comment by leehambley** [comment]
Struggling with the same, Jekyll Plus is running in a project where Grunt periodically copies files over, and very often they're the same.

**Comment by bcole808** [comment]
I am also seeing this issue. 

Steps to re-create in my project:

1. Run `guard`
2. Add a new file to the `img` folder 
3. Error is thrown: 

```
20:49:40 - ERROR - Guard::JekyllPlus failed to achieve its <run_on_additions>, exception was:
> [#] ArgumentError: same file: img/my_video.8068.46852014081.aac and img/my_video.8068.46852014081.aac
> [#] /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:1551:in `block in fu_each_src_dest'
> [#] /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:1568:in `fu_each_src_dest0'
> [#] /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:1550:in `fu_each_src_dest'
> [#] /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:393:in `cp'
(More lines omitted)
```

**Comment by imathis** [comment]
I'm about to release an update with #37, but if I can squeeze in a fix for this, I'd love it. Any ideas what's going on here @e2?

**Comment by e2** [comment]
Fixed in #39

**Comment by imathis** [comment]
I've released 2.0.2 which should address this. Thanks for reporting the issue and to @e2 for the fix!

**Comment by e2** [comment]
@imathis - no problem. Thanks for the release.
